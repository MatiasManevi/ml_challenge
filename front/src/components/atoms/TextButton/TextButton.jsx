import React from 'react';

import './TextButton.scss';

const TextButton = ({ clickHandler, text }) => {
	return (
		<a onClick={clickHandler} className="text-button">
			{text}
		</a>
	);
};

export default TextButton;
