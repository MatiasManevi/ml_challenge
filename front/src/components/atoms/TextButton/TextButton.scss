@import 'assets/scss/variables';

.text-button {
	font-size: $font-size-14;
	color: $color-neutral;
	cursor: pointer;
}
