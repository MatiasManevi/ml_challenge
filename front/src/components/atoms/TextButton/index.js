export { default } from 'components/atoms/TextButton/TextButton';
