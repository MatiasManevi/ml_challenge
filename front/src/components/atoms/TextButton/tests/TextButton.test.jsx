import React from 'react';
import { shallow } from 'enzyme';

import TextButton from '../TextButton';

describe('TextButton', () => {
	const clickHandler = jest.fn();
	const text = 'Bread';

	it('should match snapshot', () => {
		const wrapper = shallow(<TextButton clickHandler={clickHandler} text={text} />);
		expect(wrapper).toMatchSnapshot();
	});

	it('should call changeHandler when onChange', () => {
		const wrapper = shallow(<TextButton clickHandler={clickHandler} text={text} />);
		wrapper.simulate('click');
		expect(clickHandler).toHaveBeenCalledTimes(1);
	});
});
