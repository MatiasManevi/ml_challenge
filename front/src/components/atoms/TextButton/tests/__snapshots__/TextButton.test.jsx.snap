// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextButton should match snapshot 1`] = `
<a
  className="text-button"
  onClick={[MockFunction]}
>
  Bread
</a>
`;
